internal/command/secrets/deploy.go

```diff
@@ -0,0 +1,69 @@
+package secrets
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/superfly/flyctl/client"
+	"github.com/superfly/flyctl/flaps"
+	"github.com/superfly/flyctl/internal/appconfig"
+	"github.com/superfly/flyctl/internal/command"
+	"github.com/superfly/flyctl/internal/flag"
+	"github.com/superfly/flyctl/internal/flyerr"
+)
+
+func newDeploy() (cmd *cobra.Command) {
+	const (
+		long  = `Deploy staged secrets for an application`
+		short = long
+		usage = "deploy [flags]"
+	)
+
+	cmd = command.New(usage, short, long, runDeploy, command.RequireSession, command.RequireAppName)
+
+	flag.Add(cmd,
+		flag.App(),
+		flag.AppConfig(),
+		flag.Detach(),
+	)
+
+	return cmd
+}
+
+func runDeploy(ctx context.Context) (err error) {
+	client := client.FromContext(ctx).API()
+	appName := appconfig.NameFromContext(ctx)
+	app, err := client.GetAppCompact(ctx, appName)
+	if err != nil {
+		return err
+	}
+
+	if app.PlatformVersion != appconfig.MachinesPlatform {
```

Comment:
btw, you can replace all this with `command.IsMachinesPlatform(ctx, appName)` from `github.com/superfly/flyctl/internal/command` package.

Reply:
looked into this and you could do it that way and clean up the error, but then I'd had to retrieve the app twice.

Reply:
Oh right, but the thing is creating a flaps instance doesn't require the app, just the appName:

https://github.com/superfly/flyctl/blob/8ff263df70d3755bd1cc1fa10a9fa2ab65997c1c/flaps/flaps.go#L41-L43